MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -447,6 +453,416 @@ void EnggDiffractionPresenter::processRebinMultiperiod() {
+      // the correct fit function
+      // FocusedWSName is not going to change as its always going to be from
+      // single workspace
+      runEvaluateFunctionAlg(Bk2BkExpFunctionStr, FocusedWSName,
+                             current_peak_out_WS, startX, endX);
+
+      // crop workspace so only the correct workspace index is plotted
+      runCropWorkspaceAlg(current_peak_out_WS);
+
+      // apply the same binning as a focused workspace
+      runRebinToWorkspaceAlg(current_peak_out_WS);
+
+      // if the first peak
+      if (i == size_t(0)) {
+        auto renameWs =
+            Mantid::API::AlgorithmManager::Instance().createUnmanaged(
+                "RenameWorkspace");
+        g_log.notice() << "EvaluateFunction algorithm has started" << std::endl;
+        try {
+          renameWs->initialize();
+
+          renameWs->setProperty("InputWorkspace",
+                                "engggui_fitting_single_peaks0");
+          renameWs->setProperty("OutputWorkspace", single_peak_out_WS);
+          renameWs->execute();
+        } catch (std::runtime_error &re) {
+          g_log.error() << "Could not run the algorithm EvaluateFunction, "
+                           "Error description: " +
+                               static_cast<std::string>(re.what())
+                        << std::endl;
+        }
+      } else {
+        // append all peaks in to single workspace & remove
+        runAppendSpectraAlg(single_peak_out_WS, current_peak_out_WS);
+        ADS.remove(current_peak_out_WS);
+      }
+    }
+  }
+
+  m_fittingFinishedOK = true;
+}
+
+std::string EnggDiffractionPresenter::functionStrFactory(
+    Mantid::API::ITableWorkspace_sptr &paramTableWS, std::string tableName,
+    size_t row, std::string &startX, std::string &endX) {
+  AnalysisDataServiceImpl &ADS = Mantid::API::AnalysisDataService::Instance();
+  paramTableWS = ADS.retrieveWS<ITableWorkspace>(tableName);
+
+  double A0 = paramTableWS->cell<double>(row, size_t(1));
+  double A1 = paramTableWS->cell<double>(row, size_t(3));
+  double I = paramTableWS->cell<double>(row, size_t(13));
+  double A = paramTableWS->cell<double>(row, size_t(7));
+  double B = paramTableWS->cell<double>(row, size_t(9));
+  double X0 = paramTableWS->cell<double>(row, size_t(5));
+  double S = paramTableWS->cell<double>(row, size_t(11));
+
+  startX = boost::lexical_cast<std::string>(X0 - (3 * S));
+  endX = boost::lexical_cast<std::string>(X0 + (3 * S));
+
+  std::string functionStr =
+      "name=LinearBackground,A0=" + boost::lexical_cast<std::string>(A0) +
+      ",A1=" + boost::lexical_cast<std::string>(A1) +
+      ";name=BackToBackExponential,I=" + boost::lexical_cast<std::string>(I) +
+      ",A=" + boost::lexical_cast<std::string>(A) + ",B=" +
+      boost::lexical_cast<std::string>(B) + ",X0=" +
+      boost::lexical_cast<std::string>(X0) + ",S=" +
+      boost::lexical_cast<std::string>(S);
+
+  return functionStr;
+}
```

Comment:
This looks very good, and simple to extend if we need different functions, etc. in the future.